### PR TITLE
score_genes: use_raw=None instead of False, efficiency updates, prettier logging

### PR DESCRIPTION
**Pull request by falexwolf**

### CODE REVIEW

**Comment by falexwolf** [comment]
I tested this. 🙂

File: scanpy/tools/_score_genes.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def score_genes(
     score_name: str = 'score',
     random_state: Optional[Union[int, RandomState]] = 0,
     copy: bool = False,
-    use_raw: bool = False,
+    use_raw: bool = None,
```
Comment (flying-sheep) [inline scanpy/tools/_score_genes.py:23]: This is now `Optional[bool]`

File: scanpy/tools/_score_genes.py [hunk 2]
```diff
@@ -86,15 +95,17 @@ def score_genes(
     # Basically we need to compare genes against random genes in a matched
     # interval of expression.
 
+    if use_raw is None:
+        use_raw = True if adata.raw is not None else False
```
Comment (flying-sheep) [inline scanpy/tools/_score_genes.py:99]:
```py
use_raw = adata.raw is not None
```

**Review by flying-sheep — COMMENTED** [review]
Please give us time to review, now I have to do #928

**Pull request merged**